### PR TITLE
Upgrade Ember CLI 2.16.2

### LINKED ISSUE #148: Travis and yarn issue on Ember 2.12 only
I'm unsure what is special about Ember 2.12, but that case is consistently failing across all my addons while all the other cases pass.

https://travis-ci.org/shipshapecode/ember-contact-form/builds/275019161
https://travis-ci.org/shipshapecode/ember-math-helpers/builds/274758588
https://travis-ci.org/shipshapecode/ember-vivus/builds/274420286

### PR DESCRIPTION
**Pull request by juwara0**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [X] #major# - incompatible API change

# CHANGELOG
* **Updated** testing dependencies
* **Updated** to Ember CLI 2.16.2 and babel 6
* **Updated** to using ember-decorators which replaces ember-computed-decorators
* **Updated** dependencies
* **Updated** pr-bumber to version 3
* **Updated** CONTRIBUTING.md file
* **Updated** to using Ember Javascript Modules API https://github.com/ember-cli/ember-rfc176-data
* **Updated** blueprints to latest versions of dependencies
* **Updated** to use chrome headless in Travis CI
* **Updated** to using Node 8.1.2 NPM 5
* **Added** eslint-plugin-ember to enforce Ember Javascript Modules API syntax
* **Removed** running of code coverage until issue is resolved with ember-cli-code-coverage: https://github.com/kategengler/ember-cli-code-coverage/issues/133
* **Removed** running of ember-try its-2.12 scenario until issue is resolved: https://github.com/ember-cli/ember-try/issues/148
* **Updated** blueprint helper tests to be commented out since in babel 6 CommonJS files are converted to AMD (which needs import of require) and then being transpiled.

### CODE REVIEW

File: addon/components/frost-select-dropdown.js [hunk 1]
```diff
@@ -523,7 +529,7 @@ export default Component.extend({
 
         case TAB:
           this.get('onClose')()
-          return
+          return // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline addon/components/frost-select-dropdown.js:532]: Remove the `return`

File: tests/dummy/app/pods/components/demo-palette/component.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 /* global $ */
```
Comment (notmessenger) [inline tests/dummy/app/pods/components/demo-palette/component.js]: Remove and import from `jquery`?

File: tests/helpers/module-for-acceptance.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import destroyApp from '../helpers/destroy-app'
```
Comment (notmessenger) [inline tests/helpers/module-for-acceptance.js]: Don't need this file, right?

File: tests/unit/components/frost-radio-button-test.js [hunk 4]
```diff
@@ -14,14 +13,14 @@ describe(test.label, function () {
   beforeEach(function () {
     component = this.subject({
       _setupAssertions: function () {
-        return
+        return // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-radio-button-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 5]
```diff
@@ -18,18 +17,18 @@ describe(test.label, function () {
...
     ).to.equal(true)
   })
 
-  it('registers and unregisters "ps-scroll-up" event handlers', function () {
+  it('should register and unregister "ps-scroll-up" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollUp', function () { return })
+    component.set('onScrollUp', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 6]
```diff
@@ -55,11 +54,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-scroll-down" event handlers', function () {
+  it('should register and unregister "ps-scroll-down" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollDown', function () { return })
+    component.set('onScrollDown', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 7]
```diff
@@ -85,11 +84,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-y-reach-start" event handlers', function () {
+  it('should register and unregister "ps-y-reach-start" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollYStart', function () { return })
+    component.set('onScrollYStart', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: `return undefined` instead?

File: tests/unit/components/frost-scroll-test.js [hunk 8]
```diff
@@ -115,11 +114,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-y-reach-end" event handlers', function () {
+  it('should register and unregister "ps-y-reach-end" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollYEnd', function () { return })
+    component.set('onScrollYEnd', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 9]
```diff
@@ -145,11 +144,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-scroll-x" event handlers', function () {
+  it('should register and unregister "ps-scroll-x" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollX', function () { return })
+    component.set('onScrollX', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 10]
```diff
@@ -205,11 +204,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-scroll-left" event handlers', function () {
+  it('should register and unregister "ps-scroll-left" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollLeft', function () { return })
+    component.set('onScrollLeft', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 11]
```diff
@@ -235,11 +234,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "mouseenter" event handlers', function () {
+  it('should register and unregister "mouseenter" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onMouseEnter', function () { return })
+    component.set('onMouseEnter', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/components/frost-scroll-test.js [hunk 12]
```diff
@@ -175,11 +174,11 @@ describe(test.label, function () {
     $.fn.off.restore()
   })
 
-  it('registers and unregisters "ps-scroll-right" event handlers', function () {
+  it('should register and unregister "ps-scroll-right" event handlers', function () {
     const spyOn = sinon.spy($.fn, 'on')
     const spyOff = sinon.spy($.fn, 'off')
 
-    component.set('onScrollRight', function () { return })
+    component.set('onScrollRight', function () { return }) // eslint-disable-line no-useless-return
```
Comment (notmessenger) [inline tests/unit/components/frost-scroll-test.js]: Remove the `return`

File: tests/unit/lib/utils/consumer.js [hunk 13]
```diff
@@ -1,19 +1,19 @@
-const consumer = require('../../../../lib/utils/consumer')
-const expect = require('chai').expect
+// const consumer = require('../../../../lib/utils/consumer')
```
Comment (notmessenger) [inline tests/unit/lib/utils/consumer.js:1]: Get rid of the file instead of commenting out?  Or use `.skip()`?

File: tests/unit/lib/utils/npm.js [hunk 14]
```diff
@@ -1,125 +1,125 @@
...
-        {name: 'pkg2', target: '0.0.2'}
-      ])
-
-      return promise.then((result) => {
-        expect(result).to.eql('npm install --save-dev pkg1@0.0.1 pkg2@0.0.2')
-      })
-    })
-  })
-})
+// const expect = require('chai').expect
```
Comment (notmessenger) [inline tests/unit/lib/utils/npm.js:1]: Get rid of the file instead of commenting out?  Or use `.skip()`?

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: tests/dummy/app/pods/components/demo-palette/component.js (hunk 2 above)
Comment (juwara0) [reply tests/dummy/app/pods/components/demo-palette/component.js]: Fixed

File: tests/helpers/module-for-acceptance.js (hunk 3 above)
Comment (juwara0) [reply tests/helpers/module-for-acceptance.js]: Removed

File: tests/unit/components/frost-radio-button-test.js (hunk 4 above)
Comment (juwara0) [reply tests/unit/components/frost-radio-button-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 5 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 6 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 7 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Removed.

File: tests/unit/components/frost-scroll-test.js (hunk 8 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 9 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 12 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 10 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

File: tests/unit/components/frost-scroll-test.js (hunk 11 above)
Comment (juwara0) [reply tests/unit/components/frost-scroll-test.js]: Done

**Review by notmessenger — APPROVED** [review]

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/503/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: addon/components/frost-select-dropdown.js (hunk 1 above)
Comment (juwara0) [reply addon/components/frost-select-dropdown.js:532]: I will open an issue for us to refactor this code

Comment (juwara0) [reply addon/components/frost-select-dropdown.js:532]: Issue opened to track this: https://github.com/ciena-frost/ember-frost-core/issues/506

**Comment by juwara0** [comment]
It was necessary to temporarily not run ember-try scenario `ember-lts-2.12` due to an issue in CI. An issue was opened to track this: https://github.com/ciena-frost/ember-frost-core/issues/505

**Pull request merged**